tests/src/runtimeApi/memory/hipMemcpyNegativeTests.cpp

```diff
@@ -30,24 +30,25 @@ int main() {
     int* A;
     int* Ad;
     int* Bd;
-    
+
     // Allocation
     HIPCHECK(hipMalloc((void**)&Ad, sizeof(int)));
     HIPCHECK(hipMalloc((void**)&Bd, sizeof(int)));
     HIPCHECK(hipHostMalloc((void**)&A,sizeof(int)));
 
-    // Kind should be ignored and test should pass even for incorrect kind
-    HIPCHECK(hipMemcpy(Ad, A, sizeof(int), hipMemcpyDeviceToHost));
-    HIPCHECK(hipMemcpy(A,  Ad, sizeof(int), hipMemcpyHostToDevice));
-    HIPCHECK(hipMemcpy(Ad, Bd, sizeof(int), hipMemcpyHostToHost));
-    HIPCHECK(hipMemcpy(A,  A, sizeof(int), hipMemcpyDeviceToDevice));
-    
+    // If the passed pointers do not match the kind, we should return a
+    // hipErrorInvalidMemcpyDirection error
+    HIPASSERT(hipMemcpy(Ad, A, sizeof(int), hipMemcpyDeviceToHost) == hipErrorInvalidMemcpyDirection);
+    HIPASSERT(hipMemcpy(A,  Ad, sizeof(int), hipMemcpyHostToDevice) == hipErrorInvalidMemcpyDirection);
+    HIPASSERT(hipMemcpy(Ad, Bd, sizeof(int), hipMemcpyHostToHost) == hipErrorInvalidMemcpyDirection);
```

Comment:
On nvcc this test passes may be we can hide it for nvcc path and enable all tests for both nvcc and hip